### DISCUSSION TITLE
2.0: Missing parentheses in column expression using json + astext?

### DISCUSSION THREAD

**Discussion by cqfd** [sqlalchemy/sqlalchemy#10478] [Usage Questions] [answered]
Hi everyone, posting this as a discussion since I'm not sure if it's a regression in 2.0 or just a change in behavior.

In our 1.4 postgres codebase we have a few places where we do things of the form `SomeTable.json_column["foo"].astext + " " + SomeTable.json_column["bar"].astext` inside of queries. This ends up producing a fragment of sql like `(some_table.json_column ->> 'foo') || ' ' || (some_table.json_column ->> 'bar')`, which works as expected.

In 2.0, this seems to no longer work; the parentheses that surround the use of `->>` have disappeared.

Here's a snippet showing what I mean:

```
class Example(Base):
    __tablename__ = "examples"
    example_id: Mapped[int] = mapped_column(init=False, primary_key=True)
    stuff: Mapped[dict[str, Any]] = mapped_column(JSONB, nullable=False)

...

def test_example(session: Session) -> None:
    e = Example(stuff={"foo": "bar", "baz": "quux"})
    session.add(e)

    expr = Example.stuff["foo"].astext + " " + Example.stuff["baz"].astext

    # This line will fail with
    # sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedFunction) operator does not exist: text ->> unknown 
    # LINE 1: ... examples.stuff ->> 'foo' || ' ' || examples.stuff ->> 'baz' A... 
    results = session.execute(select(expr)).all()

    # desired result
    assert results == [("bar quux",)]
```

**Comment by zzzeek** [comment, marked as answer]
that's a regression caused by 428262a2d5374613f4a4cf925bbd9e94e0e.  will be fixed in 2.0.23

**Reply by zzzeek** [reply]
#10479
